Jormungand.DiscordReader-Mouth/src/main/java/events/JorgMessageReceived.java

```diff
@@ -2,20 +2,23 @@
 
 import entities.JorgUser;
 import net.dv8tion.jda.api.entities.ChannelType;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+
 import java.util.List;
 
 public class JorgMessageReceived {
-    String rawMessageContent;
-    List<JorgUser> mentions;
-    String channelId;
-    ChannelType channelType;
-    JorgUser author;
+    private String rawMessageContent;
+    private List<Member> mentions;
+    private String channelId;
+    private ChannelType channelType;
+    private JorgUser author;
 
-    public JorgMessageReceived(String rawMessageContent, List mentions, String channelId, ChannelType channelType, JorgUser author) {
-        this.rawMessageContent = rawMessageContent;
-        this.mentions = mentions;
-        this.channelId = channelId;
-        this.channelType = channelType;
-        this.author = author;
+    public JorgMessageReceived(MessageReceivedEvent event) {
+        this.rawMessageContent = event.getMessage().getContentRaw();
+        this.mentions = event.getMessage().getMentionedMembers();
```

Comment:
since this is a member it will have the JDA api issues too

Reply:
Converted to JorgUser's